clients/src/test/java/org/apache/kafka/common/requests/RequestTestUtils.java

```diff
@@ -208,10 +209,10 @@ public static MetadataResponse metadataUpdateWith(final String clusterId,
             for (int i = 0; i < numPartitions; i++) {
                 TopicPartition tp = new TopicPartition(topic, i);
                 Node leader = nodes.get(i % nodes.size());
-                List<Integer> replicaIds = Collections.singletonList(leader.id());
+                List<Integer> replicaIds = nodes.stream().map(Node::id).collect(Collectors.toList());
                 partitionMetadata.add(partitionSupplier.supply(
                         Errors.NONE, tp, Optional.of(leader.id()), Optional.ofNullable(epochSupplier.apply(tp)),
-                        replicaIds, replicaIds, replicaIds));
+                        replicaIds, replicaIds, Collections.emptyList()));
```

Comment:
Not sure about this change, looks like a re-base mistake.  Will take a closer look.

Reply:
It's needed, otherwise, all replicas are marked as offline.